### PR TITLE
fix podman play kube doesn't override dockerfile ENTRYPOINT

### PR DESCRIPTION
**Pull request by zhangguanzhang**
fix https://github.com/containers/podman/issues/6995 https://github.com/containers/podman/issues/5730 https://github.com/containers/podman/issues/7019
Signed-off-by: zhangguanzhang <zhangguanzhang@qq.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@haircommander @mheon PTAL

**Comment by haircommander** [comment]
thanks @zhangguanzhang LGTM

**Comment by haircommander** [comment]
actually, I lied, I have a nit

File: pkg/domain/infra/abi/play.go [hunk 1]
```diff
@@ -453,12 +453,17 @@ func kubeContainerToCreateConfig(ctx context.Context, containerYAML v1.Container
...
+		containerConfig.Command = imageData.Config.Entrypoint
 	}
 	if len(containerYAML.Command) != 0 {
-		containerConfig.Command = append(containerConfig.Command, containerYAML.Command...)
-	} else if imageData != nil && imageData.Config != nil {
-		containerConfig.Command = append(containerConfig.Command, imageData.Config.Cmd...)
+		containerConfig.Command = containerYAML.Command
+	}
+	if imageData != nil && imageData.Config != nil {
+		if len(containerYAML.Args) != 0 {
```
Comment (haircommander) [inline pkg/domain/infra/abi/play.go:462]: we shouldn't have this block here. it should be outside of the imageData block. We should set it to containerYAML.Args regardless of whether imageData is defined.

Comment (zhangguanzhang) [reply pkg/domain/infra/abi/play.go:462]: ok,I will change it

File: pkg/domain/infra/abi/play.go [hunk 2]
```diff
@@ -453,11 +453,14 @@ func kubeContainerToCreateConfig(ctx context.Context, containerYAML v1.Container
 
 	containerConfig.Command = []string{}
 	if imageData != nil && imageData.Config != nil {
-		containerConfig.Command = append(containerConfig.Command, imageData.Config.Entrypoint...)
+		containerConfig.Command = imageData.Config.Entrypoint
 	}
 	if len(containerYAML.Command) != 0 {
-		containerConfig.Command = append(containerConfig.Command, containerYAML.Command...)
-	} else if imageData != nil && imageData.Config != nil {
+		containerConfig.Command = containerYAML.Command
```
Comment (TomSweeneyRedHat) [inline pkg/domain/infra/abi/play.go:459]: should this still be an append, or do we want to take the yaml over the config data?  I think this is correct as is, but just wanted to double check.

Comment (zhangguanzhang) [reply pkg/domain/infra/abi/play.go:459]: pod's command should override the image's Entrypoint
pod's args should override the image's Command

Comment (zhangguanzhang) [reply pkg/domain/infra/abi/play.go:459]: could see this doc
https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes

| Description	| Docker field name	| Kubernetes field name	|
| :----- | :----- | :----- |
|The command run by the container	| Entrypoint	| command	|
|The arguments passed to the command	| Cmd	| args |

**Comment by zhangguanzhang** [comment]
Please do not merge until I add the following unit tests tomorrow

**Comment by zhangguanzhang** [comment]
all has done

**Comment by rhatdan** [comment]
LGTM
Thanks @zhangguanzhang

**Comment by zhangguanzhang** [comment]
@mheon @haircommander @TomSweeneyRedHat  PTAL

File: pkg/domain/infra/abi/play.go [hunk 3]
```diff
@@ -453,11 +453,16 @@ func kubeContainerToCreateConfig(ctx context.Context, containerYAML v1.Container
...
+	}
+	// doc https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes
+	if len(containerYAML.Args) != 0 {
+		containerConfig.Command = append(containerConfig.Command, containerYAML.Args...)
+	} else if len(containerYAML.Command) == 0 {
+		// Only containerYAML.Command can be appended when it is equals 0
```
Comment (haircommander) [inline pkg/domain/infra/abi/play.go:465]: this comment seems a bit off. I'd recommend:
`Add the Cmd from the image config only if containerYAML.Command and containerYAML.Args are empty`

**Comment by haircommander** [comment]
sorry, another nit, then LGTM

File: pkg/domain/infra/abi/play.go (hunk 3 above)
Comment (zhangguanzhang) [reply pkg/domain/infra/abi/play.go]: I have submitted the modification, and I will wait for the successful execution of CI later

**Comment by rhatdan** [comment]
/lgtm
/hold
Thanks @zhangguanzhang

**Comment by zhangguanzhang** [comment]
I think there is still a missing unit test for `ENTRYPOINT` and I will try to submit it tomorrow. Do not merge until then

**Comment by zhangguanzhang** [comment]
now all has done

**Comment by rhatdan** [comment]
Thanks @zhangguanzhang 
Nice work
/lgtm

**Pull request merged**